ROADMAP.md

```diff
@@ -0,0 +1,198 @@
+# Context
+
+> Testground is a platform for testing, benchmarking, and simulating distributed and peer-to-peer systems at scale.
+> It's designed to be multi-lingual and runtime-agnostic, scaling gracefully from 2 to 10k instances when needed.
+
+Testground was used successfully at Protocol Labs
+
+- TODO: note on libp2p interop
+- TODO: note on filecooin
+- TODO: note on IPFS v0.5.0 (massive DHT and Bitswap improvements)
+- TODO: note on libp2p (gossipsub 1.1 security extensions)
+
+# Vision
+
+Testground As A Service,
+
+- Related Products
+    - https://cockroachdb.github.io/pebble/?max=local
+    - https://perf.webkit.org/v3/
+
+
+# Problems we focus on
+
+## Table of Content
+
+  * [1. Testground provides reliable results](#1-testground-provides-reliable-results)
+  * [2. Testground can be set up as an organization-wide Service](#2-testground-can-be-set-up-as-an-organization-wide-service)
+  * [3. The knowledge required to work efficiently with Testground is available and easy to access](#3-the-knowledge-required-to-work-efficiently-with-testground-is-available-and-easy-to-access)
+  * [4. Testground Developement follows high-standard](#4-testground-developement-follows-high-standard)
+  * [5. Testground provides the networking tooling required to test complex Distributed / Decentralized Applications](#5-testground-provides-the-networking-tooling-required-to-test-complex-distributed---decentralized-applications)
+  * [6. Testground provides the tooling to make identifying and fixing issues easy.](#6-testground-provides-the-tooling-to-make-identifying-and-fixing-issues-easy)
+  * [7. Testground covers every /essential/ combinations of languages and libraries required by its users](#7-testground-covers-every--essential--combinations-of-languages-and-libraries-required-by-its-users)
+
+
+## 1. Testground provides reliable results
+
+**Why:** A testing platform that is unreliable is just a noise-producing machine. We need to secure our users' trust. Testground maintainers need clear feebdacks about stability improvements & regressions.
+
+### Milestone 1: We have a stability metrics dashboard
+
+Maintainers and users have a way to measure and follow Testground's Stability over any "relevant" axis.
+
+This might combine different languages, runners (k8s, docker, local), and context (developer env, CI env, k8s env).
+
+
+### Milestone 2: Every known stability issue has been solved
+
+- local runner,
+- docker runner,
+- EKS runner,
+
+
+### Milestone 3: We have a performances dashboard
```

Comment:
I don't know: Testground performance is a topic we care about and we'll focus on it after we're done with stability, but deciding which metrics we measure, and under which conditions, is part of the problem.

Reply:
(I updated the doc with a note)

---

Maintainers and users have a way to measure and follow Testground's Performances over any "relevant" axis. This effort will start with identifying which metrics we want to measure first.

It might contain:

- Raw build / run performance for synthetic workload
- Performance of real-life usage in CI (like `libp2p/test-plans`)
- Consideration for caching features, re-building, etc.


---